admin_manual/installation/nginx.rst

```diff
@@ -205,71 +209,52 @@ In this example the webroot is located at
-      # Before enabling Strict-Transport-Security headers please read into this
-      # topic first.
-      #add_header Strict-Transport-Security "max-age=15768000; includeSubDomains; preload;" always;
-      #
+      # HSTS settings
       # WARNING: Only add the preload option once you read about
       # the consequences in https://hstspreload.org/. This option
       # will add the domain to a hardcoded list that is shipped
       # in all major browsers and getting removed from this list
       # could take several months.
-      add_header Referrer-Policy "no-referrer" always;
-      add_header X-Content-Type-Options "nosniff" always;
-      add_header X-Download-Options "noopen" always;
-      add_header X-Frame-Options "SAMEORIGIN" always;
-      add_header X-Permitted-Cross-Domain-Policies "none" always;
-      add_header X-Robots-Tag "none" always;
-      add_header X-XSS-Protection "1; mode=block" always;
-
-      # Remove X-Powered-By, which is an information leak
-      fastcgi_hide_header X-Powered-By;
-
-      # Path to the root of your installation
-      root /var/www;
-
+      #add_header Strict-Transport-Security "max-age=15768000; includeSubDomains; preload;" always;
+      
       location = /robots.txt {
           allow all;
           log_not_found off;
           access_log off;
       }
 
-      # The following 2 rules are only needed for the user_webfinger app.
-      # Uncomment it if you're planning to use this app.
-      #rewrite ^/.well-known/host-meta /nextcloud/public.php?service=host-meta last;
-      #rewrite ^/.well-known/host-meta.json /nextcloud/public.php?service=host-meta-json last;
+      location /.well-known {
+          # The following 6 rules are borrowed from `.htaccess`
 
-      # The following rule is only needed for the Social app.
-      # Uncomment it if you're planning to use this app.
-      #rewrite ^/.well-known/webfinger /nextcloud/public.php?service=webfinger last;
+          rewrite ^/\.well-known/host-meta\.json  /nextcloud/public.php?service=host-meta-json    last;
+          rewrite ^/\.well-known/host-meta        /nextcloud/public.php?service=host-meta         last;
+          rewrite ^/\.well-known/webfinger        /nextcloud/public.php?service=webfinger         last;
+          rewrite ^/\.well-known/nodeinfo         /nextcloud/public.php?service=nodeinfo          last;
 
-      location = /.well-known/carddav {
-        return 301 $scheme://$host:$server_port/nextcloud/remote.php/dav;
-      }
-      location = /.well-known/caldav {
-        return 301 $scheme://$host:$server_port/nextcloud/remote.php/dav;
-      }
-
-      location /.well-known/acme-challenge { }
-
-      location ^~ /nextcloud {
+          location = /.well-known/carddav   { return 301 /nextcloud/remote.php/dav/; }
+          location = /.well-known/caldav    { return 301 /nextcloud/remote.php/dav/; }
 
+          try_files $uri $uri/ /nextcloud/index.php$request_uri;
```

Comment:
What is this for? If a client is requesting a URL path not under /nextcloud, or another Nextcloud related path like .well-known/caldav, this might not even be for Nextcloud.
For example, in my server config I have different /subdirs for different PHP application. Maybe we should just let NGINX serve a 404 instead of leading any request to Nextcloud.

Reply:
I was under the impression that NextCloud plugins/apps might want to use other well-known URIs, and the front-end controller would handle those. If you know this not to be the case, or still don't want to fallback to NextCloud controller, we can remove this. It is probably the right thing to do.

Reply:
We should also probably use the same line in the "Nextcloud in webroot" config, i.e. `try_files $uri $uri/ =404`.